cmd/metal-api/internal/service/machine-service.go

```diff
@@ -1820,7 +1820,7 @@ func ResurrectMachines(ds *datastore.RethinkStore, publisher bus.Publisher, ep *
 			continue
 		}
 
-		if provisioningEvents.Liveliness == metal.MachineLivelinessDead && time.Since(*provisioningEvents.LastEventTime) > metal.MachineResurrectAfter {
+		if provisioningEvents.Liveliness == metal.MachineLivelinessDead && time.Since(*provisioningEvents.LastEventTime) > metal.MachineResurrectAfter && m.State.Value != metal.ShutdownState {
```

Comment:
It's easier to understand moving this condition further up in the loop and guard code with `continue` as done in the lines above.